sklearn/_fast_gradient_boosting/gradient_boosting.py

```diff
@@ -0,0 +1,754 @@
+"""Fast Gradient Boosting decision trees for classification and regression."""
+from abc import ABC, abstractmethod
+
+import numpy as np
+from time import time
+from sklearn.base import BaseEstimator, RegressorMixin, ClassifierMixin
+from sklearn.utils import check_X_y, check_random_state, check_array
+from sklearn.utils.validation import check_is_fitted
+from sklearn.utils.multiclass import check_classification_targets
+from sklearn.metrics import check_scoring
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import LabelEncoder
+from ._gradient_boosting import _update_raw_predictions
+from .types import Y_DTYPE, X_DTYPE, X_BINNED_DTYPE
+
+from .binning import BinMapper
+from .grower import TreeGrower
+from .loss import _LOSSES
+
+
+class BaseFastGradientBoosting(BaseEstimator, ABC):
+    """Base class for fast gradient boosting estimators."""
+
+    @abstractmethod
+    def __init__(self, loss, learning_rate, n_estimators, max_leaf_nodes,
+                 max_depth, min_samples_leaf, l2_regularization, max_bins,
+                 scoring, validation_fraction, n_iter_no_change, tol, verbose,
+                 random_state):
+        self.loss = loss
+        self.learning_rate = learning_rate
+        self.n_estimators = n_estimators
+        self.max_leaf_nodes = max_leaf_nodes
+        self.max_depth = max_depth
+        self.min_samples_leaf = min_samples_leaf
+        self.l2_regularization = l2_regularization
+        self.max_bins = max_bins
+        self.n_iter_no_change = n_iter_no_change
+        self.validation_fraction = validation_fraction
+        self.scoring = scoring
+        self.tol = tol
+        self.verbose = verbose
+        self.random_state = random_state
+
+    def _validate_parameters(self):
+        """Validate parameters passed to __init__.
+
+        The parameters that are directly passed to the grower are checked in
+        TreeGrower."""
+
+        if self.loss not in self._VALID_LOSSES:
+            raise ValueError(
+                "Loss {} is not supported for {}. Accepted losses: "
+                "{}.".format(self.loss, self.__class__.__name__,
+                             ', '.join(self._VALID_LOSSES)))
+
+        if self.learning_rate <= 0:
+            raise ValueError('learning_rate={} must '
+                             'be strictly positive'.format(self.learning_rate))
+        if self.n_estimators < 1:
+            raise ValueError('n_estimators={} must not be smaller '
+                             'than 1.'.format(self.n_estimators))
+        if self.n_iter_no_change is not None and self.n_iter_no_change < 0:
+            raise ValueError('n_iter_no_change={} must be '
+                             'positive.'.format(self.n_iter_no_change))
+        if (self.validation_fraction is not None and
```

Comment:
No need to because unlike `validation_fraction` of `ensemble.gradient_boosting` this can also be an int to specify the absolute validation set size (instead of a proportion).

Thanks for noticing this I'll edit the *changed parameters* section